### ISSUE TITLE
Referring params returning empty json object

### ISSUE THREAD

**Issue by alokomkar** [BranchMetrics/android-branch-deep-linking-attribution#669]
Hi,

We are working with Branch's deeplinking to perform screen redirect in our app. This is a working code as we have tested this out before : 

```
Branch: Latest Referring Params: {
        "+clicked_branch_link": false,
        "+is_first_session": false
    }
```
We are expecting an additional parameter : deepLinkPageName with the page name value to appear along with the default parameters, we went through the issues list and have found that this has been encountered before and was solved by updating to latest version of branch sdk; we currently are using 3.0.2 version.

We would appreciate assistance in this matter. 

Thanks.

**Comment by csalmi-branch** [comment]
Hi @alokomkar! Do you still have this issue if you use 3.0.4?

**Comment by MoeMamdouh** [comment]
i user 3.0.4 and still params returning empty, any help please? @csalmi-branch

**Comment by csalmi-branch** [comment]
@MoeMamdouh Can you confirm for me that you do not have a key mistmatch? The key you use to create the link must be the same key in the app (if you use your test key to create the link, use your test key in the configuration of the app).

**Comment by heliomartins** [comment]
I'm getting the same issue. My scenario is:
1. SDK Version is: 3.0.4;
2. I Have defined in my AndroidManifest.xml Branch App Links:

```          
            <intent-filter android:autoVerify="true">
                <action android:name="android.intent.action.VIEW" />
                <category android:name="android.intent.category.DEFAULT" />
                <category android:name="android.intent.category.BROWSABLE" />
                <data android:scheme="https" android:host="xxx.app.link" />
                <data android:scheme="https" android:host="xxx.test-app.link" />
            </intent-filter>
```

Scenario to Reproduce:
**It works when:**
- When the user enter the deeplink in a browser.
- The APP must have called the endpoint: https://api2.branch.io/v1/close

**It doesn't work when:**
- The user clicks in the deeplink from inside the APP.
- The close endpoint: https://api2.branch.io/v1/close unless we explicitly put the app in Background.
- We have tried to use the feature that had been released Branch.forceEnabledSession. but it didn't affect the result.

**Comment by apeterson-branch** [comment]
@heliomartins Android v3.1.1 specifically addressed issues where the app is in the foreground and receiving the link -- can you please let me know if this addresses your issue?

**Comment by heliomartins** [comment]
Thanks for reaching back @apeterson-branch. I will make some tests and let you know soon.

**Comment by shuzootani** [comment]
Hallo @apeterson-branch, @csalmi-branch 
I upgraded to v3.2.0 but it still returns empty params `{ }` on foreground.
my issue happens in react side tho.

``` js
branch.subscribe(({ error, params }) => {
  // params = {}
})
```

I'm using [react-native-branch@v3.0.1](https://github.com/BranchMetrics/react-native-branch-deep-linking-attribution/blob/81c4f224d146d97f1aaae030d065bcb674f1526d/ChangeLog.md) which supports android-sdk@v3.2.0
> Uses native Branch SDKs 3.2.0 (Android), 0.27.0 (iOS).

Here is the diff for upgrading.

``` diff
build.gradle

- implementation fileTree(dir: "libs", include: ["*.jar"])

- api "io.branch.sdk.android:library:2.+"
+ api "io.branch.sdk.android:library:3.2+"
```

Let me know if I'm missing something!
Thank you

**Comment by csalmi-branch** [comment]
@jdee this seems potentially related to what you are working on. Could you take a look?

**Comment by ZalakSA** [comment]
@heliomartins Did you checked the solution provided, did it worked? I am also facing same issue for me also params returning empty.

**Comment by ZalakSA** [comment]
@csalmi-branch Please can you also help me and guide on this issue?

**Comment by csalmi-branch** [comment]
We are working internally on an issue that I believe is related to this one. As soon as we have any updates (or a fix if we find a bug), we will update this thread.

**Comment by heliomartins** [comment]
> @heliomartins Did you checked the solution provided, did it worked? I am also facing same issue for me also params returning empty.

Haven't tried yet. I've gotta some other priorities...

As soon as I get back to this problem, and make some tests, I will update if it works. Sorry about it.

**Comment by ecorengia** [comment]
Facing same problem on version 4.0.0. Referring params is being populated when the app is in background. But it does not the same on foreground. Please help. I already checked similar posts and README file.

**Comment by csalmi-branch** [comment]
This issue should be fixed starting on v4.1.0. I'm closing this issue, but please let me know if the problem still exists!

**Comment by akshaymanathkar** [comment]
I was using 2.19.5 where i had this issue . Today I updated version to 4.1.1, still seeing this issue
i have some what similar issue. In my scenario i am using Web PAI to generate Deep link and then calling redirect instead of user clicking it. IT opens the app but following are my referring params 
{
        "+clicked_branch_link": false,
        "+is_first_session": false
}

Is this issue still there for my scenario
